src/main/java/com/synopsys/integration/alert/provider/blackduck/tasks/EmailSyncTask.java

```diff
@@ -0,0 +1,102 @@
+import com.synopsys.integration.blackduck.api.generated.view.UserView;
+import com.synopsys.integration.blackduck.service.HubService;
+import com.synopsys.integration.exception.IntegrationException;
+
+@Component
+public class EmailSyncTask extends SyncTask<String> {
+
+    @Autowired
+    public EmailSyncTask(final TaskScheduler taskScheduler, final BlackDuckProperties blackDuckProperties, final BlackDuckUserRepositoryAccessor blackDuckUserRepositoryAccessor) {
+        super(taskScheduler, "blackduck-sync-email-task", blackDuckProperties, blackDuckUserRepositoryAccessor);
+    }
+
+    @Override
+    public List<UserView> getHubViews(final HubService hubService) throws IntegrationException {
+        return hubService.getAllResponses(ApiDiscovery.USERS_LINK_RESPONSE);
+    }
+
+    @Override
+    public Map<String, ? extends HubView> getCurrentData(final List<? extends HubView> hubViews) {
+        final List<UserView> userViews = (List<UserView>) hubViews;
+
+        final Map<String, ? extends HubView> emailAddressMap = new HashMap<>();
+        for (final UserView userView : userViews) {
+            if (StringUtils.isNotBlank(userView.email)) {
+                // We do not need the HubView's here because that is used for adding relations and we wont be adding relations in the EmailSyncTask
+                emailAddressMap.put(userView.email, null);
+            }
+        }
+        return emailAddressMap;
+    }
+
+    @Override
+    public Set<String> getStoredData(final List<? extends DatabaseEntity> storedEntities) {
+        final List<BlackDuckUserEntity> blackDuckUserEntities = (List<BlackDuckUserEntity>) storedEntities;
+
+        final Set<String> storedEmails = blackDuckUserEntities.stream().map(blackDuckUserEntity -> blackDuckUserEntity.getEmailAddress()).collect(Collectors.toSet());
+        return storedEmails;
+    }
+
+    @Override
+    public List<? extends DatabaseEntity> getEntitiesToRemove(final List<? extends DatabaseEntity> storedEntities, final Set<String> dataToRemove) {
+        final List<BlackDuckUserEntity> blackDuckUserEntities = (List<BlackDuckUserEntity>) storedEntities;
+        final List<BlackDuckUserEntity> blackDuckUsersToRemove = blackDuckUserEntities.stream()
+                                                                     .filter(blackDuckUserEntity -> dataToRemove.contains(blackDuckUserEntity.getEmailAddress()))
+                                                                     .collect(Collectors.toList());
+        return blackDuckUsersToRemove;
+    }
+
+    @Override
+    public DatabaseEntity createEntity(final String data) {
+        return new BlackDuckUserEntity(data, false);
+    }
+
+    @Override
+    public void addRelations(final Map<String, ? extends HubView> currentDataMap, final List<? extends DatabaseEntity> storedEntities, final HubService hubService) throws IOException, IntegrationException {
+
```

Comment:
Is this just empty because we don't have any email-only configurations yet?

Reply:
This is empty, because the Email syncing does not add relationships. The Group syncing and Project syncing do.  I can remove this from the object contract and leave an empty default method in the parent if you would prefer that